core/pkg/eval/json_evaluator.go

```diff
@@ -341,21 +341,26 @@ func (je *JSONEvaluator) evaluateVariant(reqID string, flagKey string, context m
 			je.Logger.ErrorWithID(reqID, fmt.Sprintf("error applying rules: %s", err))
 			return "", flag.Variants, model.ErrorReason, metadata, errors.New(model.ParseErrorCode)
 		}
+
+		// check if string is "null" before we strip quotes, so we can differentiate between JSON null and "null"
+		trimmed := strings.TrimSpace(result.String())
+		if trimmed == "null" {
+			return flag.DefaultVariant, flag.Variants, model.DefaultReason, metadata, nil
+		}
+
 		// strip whitespace and quotes from the variant
-		variant = strings.ReplaceAll(strings.TrimSpace(result.String()), "\"", "")
+		variant = strings.ReplaceAll(trimmed, "\"", "")
 
 		// if this is a valid variant, return it
 		if _, ok := flag.Variants[variant]; ok {
 			return variant, flag.Variants, model.TargetingMatchReason, metadata, nil
+		} else {
```

Comment:
the linter is right, this else can be removed and we can directly do the `return` here

Reply:
You are somehow commenting on an old diff. You are right that this was a problem, but I fixed it yesterday :thinking: